src/RoslynAnalyzers/Directory.Packages.props

```diff
@@ -0,0 +1,6 @@
+<Project>
+  <Import Project="..\..\Directory.Packages.props" />
+  <PropertyGroup>
+    <CentralPackageTransitivePinningEnabled>false</CentralPackageTransitivePinningEnabled>
```

Comment:
You may want to add a comment here why. Your PR description was clear, but that might be hard to find for somebody who runs into this.

Reply:
In this package shouldn't we be adding `<PackageReference>` with version overrides? Basically this is the place where we should set the value for `System.Collections.Immutable` used by every analyzer?

Reply:
I wasn't thinking about direct dependency, I was thikning about using this file to reset the versions of certain DLLs for everything below this. This is how we used to manage the razor compiler within the razor repo (it used to require older versions of MS.CA and related) than the rest of the IDE. It has all the same problems as analyzers here. 

This is how it used to look

https://github.com/dotnet/razor/blob/release/dev17.12/src/Compiler/Directory.Packages.props

Reply:
Gotcha, that looks preferable to direct dependencies. Now to figure out what all to include in there.

Reply:
Think this is the full list that we need to re-version in this file:

```
System.Buffers (>= 4.5.1)
System.Collections.Immutable (>= 9.0.0)
System.Composition (>= 9.0.0)
System.IO.Pipelines (>= 9.0.0)
System.Memory (>= 4.5.5)
System.Numerics.Vectors (>= 4.5.0)
System.Reflection.Metadata (>= 9.0.0)
System.Runtime.CompilerServices.Unsafe (>= 6.0.0)
System.Text.Encoding.CodePages (>= 7.0.0)
System.Threading.Channels (>= 7.0.0)
System.Threading.Tasks.Extensions (>= 4.5.4)
```

Got that list by just going to the [MS.CA](https://www.nuget.org/packages/Microsoft.CodeAnalysis) NuPkg entry and looking at the listed dependencies.

Reply:
Thanks @jaredpar!

Reply:
I believe we should take this change as is and work on re-enabling transitive pinning as a follow up. The blocker from doing it in one go is that we are currently using n versions of M.CA and would need to track these dependencies for each. I've been working on fixing up Jared's unify M.CA versions PR, but we would still need to account for 3 versions of M.CA and I am working through nullablility fixes moving the analyzers to 4.0.1.